File: activations.py
from math import exp, sinh, cosh


def sigmoid(activation):
    return 1.0 / (1.0 + exp(-activation))


def sigmoid_derivative(output):
    return output * (1.0 - output)


def tanhx(activation):
    return sinh(activation) / cosh(activation)


def tanh_derivative(output):
    return 1 - (output**2)


def lrelu(z, a=0.01):
    if z > 0:
        return z
    elif z <= 0:
        return z * a


def lrelu_derivative(z, a=0.01):
    if z > 0:
        return 1
    elif z <= 0:
        return a

File: java/jbnn/src/Activations.java
import java.lang.Math;
import java.util.Random;

public final class Activations {
    public static Random rand = new Random();
    public static float sigmoid(float input) {
        return 1.0f / (1.0f + (float) Math.exp(-input));
    }

    public static float tanh(float input) {
        return (float)(Math.sinh(input) / Math.cosh(input));
    }

    public static float dSigmoid(float output) {
        return output * (1.0f - output);
    }

    public static float dTanh(float output) {
        return 1.0f - (output * output);
    }
}